salt/crypt.py

```diff
@@ -822,7 +822,40 @@ def minion_sign_in_payload(self):
         if 'autosign_grains' in self.opts:
             autosign_grains = {}
             for grain in self.opts['autosign_grains']:
-                autosign_grains[grain] = self.opts['grains'].get(grain, None)
+                log.trace('Autosign Grains - grain %s - Looking for value', grain)
+                autosign_grains[grain] = None
+                # If the autosign_grain is directly accessible, use it.
+                # it avoids issues if the grain key itself contains columns
+                element = None
+                if grain in self.opts['grains']:
+                    log.trace('Autosign Grains - grain %s - Found in grain root', grain)
+                    element = self.opts['grains'].get(grain, None)
+                # Else try to split the the grain key with column
+                # In every case, if the autosign_grain could not be found anywhere, return None
+                else:
+                    current_level = self.opts['grains']
```

Comment:
wondering if this code could be condensed down to using `salt.utils.data.traverse_dict_and_list`

For example, given a grains structure that looks like this:

```
 ch3ll@thecakeisalie  ~/git/salt   autosign_grains ●  cat /etc/salt/grains 
nest:
  one:
    two:
      - this
```

```
In [6]: salt.utils.data.traverse_dict_and_list(__grains__, 'nest:one:two', delimiter=':')                                                            
Out[6]: ['this']
```

```
In [6]: salt.utils.data.traverse_dict_and_list(__grains__, 'nest:one:two', delimiter=':')                                                            
Out[6]: ['this']
```

You can look at how we are currently doing this in the grains execution module in the get method as well.